src/components/Composer/index.native.tsx

```diff
@@ -109,6 +112,15 @@ function Composer(
     const maxHeightStyle = useMemo(() => StyleUtils.getComposerMaxHeightStyle(maxLines, isComposerFullSize), [StyleUtils, isComposerFullSize, maxLines]);
     const composerStyle = useMemo(() => StyleSheet.flatten([style, textContainsOnlyEmojis ? styles.onlyEmojisTextLineHeight : {}]), [style, textContainsOnlyEmojis, styles]);
 
+    useEffect(() => {
+        if (!showSoftInputOnFocus) {
+            return;
+        }
+        setTimeout(() => {
```

Comment:
To avoid using setTimeout, I suggest using isKeyboardShown

```
if (!showSoftInputOnFocus || !isKeyboardShown) {
            return;
}
setContextMenuHidden(false);
```

@christianwen Please try this approach only after resolving [this bug](https://github.com/Expensify/App/pull/32711#issuecomment-2348360000)